gateway-ha/src/main/java/io/trino/gateway/ha/security/LbLdapClient.java

```diff
@@ -76,6 +76,7 @@ public LbLdapClient(LdapConfiguration ldapConfig) {
     poolConfig.setMaxIdle(1);
     poolConfig.setMaxTotal(1);
     poolConfig.setMinIdle(0);
+    poolConfig.setTestOnBorrow(true);
```

Comment:
I think your looking at the document for tomcat jdbc, which in this case it is rather apache commons pool2. Here is the [document](https://commons.apache.org/proper/commons-pool/apidocs/org/apache/commons/pool2/impl/BaseGenericObjectPool.html#setTestOnBorrow(boolean)) for it.

> Sets whether objects borrowed from the pool will be validated before being returned from the borrowObject() method. Validation is performed by the validateObject() method of the factory associated with the pool. If the object fails to validate, it will be removed from the pool and destroyed, and a new attempt will be made to borrow an object from the pool.

To elaborate more, `validateObject()` is depended on by the validating factory in line 80, `ValidatingPoolableLdapConnectionFactory validatingFactory = new ValidatingPoolableLdapConnectionFactory(defaultFactory);`. [ValidatingPoolableLdapConnectionFactory](https://nightlies.apache.org/directory/api/2.1.2/apidocs/org/apache/directory/ldap/client/api/ValidatingPoolableLdapConnectionFactory.html) extends [AbstractPoolableLdapConnectionFactory](https://nightlies.apache.org/directory/api/2.1.2/apidocs/org/apache/directory/ldap/client/api/AbstractPoolableLdapConnectionFactory.html#validateObject-org.apache.commons.pool2.PooledObject-), where validateObject is defined. `setValidator` explains the default validator (An implementation of [LdapConnectionValidator](https://nightlies.apache.org/directory/api/2.1.2/apidocs/org/apache/directory/ldap/client/api/LdapConnectionValidator.html) that attempts a simple lookup on the rootDSE.)